feat: floating Max v3

kappa90 opened the pull request:
> [!IMPORTANT]
> 👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Problem
We already have a floating Max v1 and v2, time for a v3.
In this version, the chat can happen directly above the floating input.

https://github.com/user-attachments/assets/91411a60-d082-4b72-a8a6-ab2118d3e6d2

## Changes
- Add a thread view above the floating input

## Did you write or update any docs for this change?

<!-- Engineers are responsible for doing the first pass at documenting their features and/or code.  -->

- [ ] I've [added or updated the docs](https://posthog.com/handbook/engineering/writing-docs)
- [ ] I've reached out for help from the docs team
- [X] No docs needed for this change

## How did you test this code?
Tested visually + stories

Commit:
feat: floating hog

Commit:
fix: move logic to kea

kappa90 committed:
Update frontend/src/scenes/max/MaxFloatingInput.tsx

Co-authored-by: graphite-app[bot] <96075541+graphite-app[bot]@users.noreply.github.com>

Commit:
fix: lint

Twixes committed:
Merge branch 'master' into feat/max-floating-input

Twixes committed:
Tweak floating input positioning

Commit:
feat: floating experiment v1

Commit:
fix: reuse components

Commit:
fix: location change bug

github-actions[bot] committed:
Update UI snapshots for `chromium` (3)

github-actions[bot] committed:
Update UI snapshots for `chromium` (5)

Commit:
fixes

Commit:
Merge remote-tracking branch 'origin/max/floating-ui-experiment' into max/floating-ui-experiment

Commit:
fix: objectsEqual throwing exception

Commit:
feat: floating max v3

Commit:
fix: autoscroll + small insights

Commit:
Merge remote-tracking branch 'origin/master' into max/floating-experiment-v3

Commit:
feat: merge v2.5

kappa90 commented:
Merged [this PR](https://github.com/PostHog/posthog/pull/33960) manually.

github-actions[bot] committed:
Update UI snapshots for `chromium` (3)

github-actions[bot] committed:
Update UI snapshots for `chromium` (6)

Commit:
fix: extra weird kea bug

Commit:
fix: small bug

Commit:
fix: tests

Commit:
fix: test

Commit:
fix: stories

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (5)

Commit:
Merge remote-tracking branch 'origin/master' into max/floating-experiment-v3

Commit:
Merge remote-tracking branch 'origin/max/floating-experiment-v3' into max/floating-experiment-v3

github-actions[bot] committed:
Update UI snapshots for `chromium` (7)

Commit:
fix: stories

Commit:
Merge remote-tracking branch 'origin/master' into max/floating-experiment-v3

github-actions[bot] committed:
Update UI snapshots for `chromium` (5)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

Commit:
fix: i'm dumb

Twixes committed:
Merge branch 'master' into max/floating-experiment-v3

github-actions[bot] committed:
Update UI snapshots for `chromium` (3)

github-actions[bot] committed:
Update UI snapshots for `chromium` (5)

github-actions[bot] committed:
Update UI snapshots for `chromium` (7)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

Commit:
fix: remove array syntax

Commit:
Merge remote-tracking branch 'origin/max/floating-experiment-v3' into max/floating-experiment-v3

Twixes committed:
Fix Max gone for users without floating flag

Twixes committed:
Restore textarea width limit

`w-full` without a maximum gets wiiide, especially in the /max scene!

github-actions[bot] committed:
Update UI snapshots for `chromium` (5)

github-actions[bot] committed:
Update UI snapshots for `chromium` (3)

github-actions[bot] committed:
Update UI snapshots for `chromium` (7)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

Twixes committed:
Fix lint from Oxlint

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

Twixes commented on frontend/src/scenes/max/components/HedgehogAvatar.tsx:
```diff
@@ -0,0 +1,75 @@
+                delayMs={0}
+            >
+                <div
+                    className="size-10 rounded-full overflow-hidden border border-border-primary shadow-lg hover:shadow-xl transition-all duration-200 cursor-pointer -scale-x-100 hover:scale-y-110 hover:-scale-x-110 flex items-center justify-center bg-bg-light"
```
These shadows are pretty un-PostHoggy, I propose we give him the same padding with translucent background as the actual input overlay (and no shadow)
![A - over](https://github.com/user-attachments/assets/ba008bda-39dc-45f6-8f8c-509245c04792)

Commit:
fix: sticky question input

Twixes commented on frontend/src/scenes/max/components/SuggestionsDisplay.tsx:
```diff
@@ -0,0 +1,149 @@
+                                size="xxsmall"
+                                type="tertiary"
+                                fullWidth
+                                disabledReason={
+                                    !dataProcessingAccepted ? 'Please accept OpenAI processing data' : undefined
+                                }
```
Let's do something slightly unusual, because we're looking at a _group_ of disabled buttons here, so it feels odd for each to have a separate, but identical disabled message.

Let's set only `disabled={!dataProcessingAccepted}` on each button, and then wrap the `<div className="flex flex-col gap-1">` in a `<Tooltip title={!dataProcessingAccepted ? 'Please accept OpenAI processing data' : undefined}>`.

Twixes committed:
Disable input when awaiting data processing approval

Twixes reviewed (changes requested):
Awesome stuff, a few comments though.
Let's restore that original thread input styling we've talked about.
Pushed a few tweaks as part of review, make sure they all make sense!

Commit:
feat: drag and snap

Commit:
fix: avatar and question inside same container

Commit:
refactor: change component names

Commit:
fix: autoscroll on thread minimize

Commit:
fix: keep context when minimizing

Commit:
fix: type

Commit:
fix: flickering stories

Commit:
Merge remote-tracking branch 'origin/max/floating-experiment-v3' into max/floating-experiment-v3

kappa90 commented on frontend/src/scenes/max/components/FloatingInputActions.tsx:
```diff
@@ -0,0 +1,72 @@
+                    }}
+                />
+            </Tooltip>
+            <Tooltip title="Minimize" placement="top" delayMs={0}>
```
Renamed to 'Close'

github-actions[bot] committed:
Update UI snapshots for `chromium` (9)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (9)

Commit:
fix: max input width

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

Commit:
fix: stories

Commit:
Merge remote-tracking branch 'origin/max/floating-experiment-v3' into max/floating-experiment-v3

Commit:
fix: style

Commit:
fix: reset snapshots

Commit:
Merge remote-tracking branch 'origin/master' into max/floating-experiment-v3

Commit:
Merge branch 'master' into max/floating-experiment-v3

Commit:
fix: merge from master

github-actions[bot] committed:
Update UI snapshots for `chromium` (8)

github-actions[bot] committed:
Update UI snapshots for `chromium` (16)

github-actions[bot] committed:
Update UI snapshots for `chromium` (11)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

Twixes committed:
Make floating appearance smoother

Generally scaling appearance animations from 0 or low values like 0.1 makes them TOO strong. Range between 0.5-0.9 – much smoother.

Twixes committed:
PostHoggify the hog

No 3D shadow.

Twixes committed:
Fix useDragAndSnap() activating on right click

Twixes committed:
Slightly fix useDragAndSnap positioning

It's still broken if navbar is collapsed.

Twixes committed:
Merge branch 'master' into max/floating-experiment-v3

Twixes commented on frontend/src/scenes/max/hooks/useDragAndSnap.ts:
```diff
@@ -0,0 +1,167 @@
+                    finalX = windowWidth - cssRightOffset - 16 - 40 // Direct positioning from right edge
+                } else {
+                    // For left side: cssLeftOffset + (containerWidth - 16 - 40)
+                    finalX = cssLeftOffset + containerWidth - 16 - 40 - 40 // Position accounting for container layout
```
This positioning are hard to understand and extremely fragile:

1. Doesn't support navbar being closed
2. Will look wrong the moment navigation margins or width are changed
3. `useDragAndSnap` positioning is calculated entirely separately, and differently, from the final states of `MaxFloatingInput`

Okay for a limited release, but will need re-engineering for release

Twixes reviewed (approved):
Looks good, let's go!

I think I liked the previous input appearance animation more, but this is okay. We're missing an exit animation though - rarely should there only be one of exit/enter.

> (I gave it an attempt to smoothly morph between hedgehog and input here, but ran into inexplicable sizing issues) 
> ![2025-06-25 18 09 00](https://github.com/user-attachments/assets/b5890ef0-fce4-40b7-bc64-1b62a1caffe6)

Twixes committed:
Fix two Max stories

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)
